sphinx_rtd_theme/versions.html

```diff
@@ -1,34 +1,47 @@
-{% if READTHEDOCS %}
-{# Add rst-badge after rst-versions for small badge style. #}
-  <div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="{{ _('Versions') }}">
+<script async type="text/javascript" src="http://localhost:8000/readthedocs-addons.js"></script>
+<script type="text/javascript">
+ document.addEventListener("readthedocs-addons-data-ready", function(event) {
+     const config = event.detail;
+
+     const flyout = `
+  <div class="rst-versions" data-toggle="rst-versions" role="note">
     <span class="rst-current-version" data-toggle="rst-current-version">
       <span class="fa fa-book"> Read the Docs</span>
-      v: {{ current_version }}
+      v: ${ config.versions.current.slug }
       <span class="fa fa-caret-down"></span>
     </span>
     <div class="rst-other-versions">
       <dl>
-        <dt>{{ _('Versions') }}</dt>
-        {% for slug, url in versions %}
-          <dd><a href="{{ url }}">{{ slug }}</a></dd>
-        {% endfor %}
+        <dt>Languages</dt>
+        ${ config.addons.flyout.translations.map(
+          (language) => `<dd><a href="${ language.url }">${ language.slug }</a></dd>`).join("\n")}
       </dl>
       <dl>
-        <dt>{{ _('Downloads') }}</dt>
-        {% for type, url in downloads %}
-          <dd><a href="{{ url }}">{{ type }}</a></dd>
-        {% endfor %}
+        <dt>Versions</dt>
+        ${ config.addons.flyout.versions.map(
+          (version) => `<dd><a href="${ version.url }">${ version.slug }</a></dd>`).join("\n")}
       </dl>
       <dl>
-        {# Translators: The phrase "Read the Docs" is not translated #}
-        <dt>{{ _('On Read the Docs') }}</dt>
+        <dt>Downloads</dt>
+        ${ config.addons.flyout.downloads.map(
+          (download) => `<dd><a href="${ download.url }">${ download.name }</a></dd>`).join("\n")}
+      </dl>
+      <dl>
+        <dt>On Read the Docs</dt>
           <dd>
-            <a href="//{{ PRODUCTION_DOMAIN }}/projects/{{ slug }}/?fromdocs={{ slug }}">{{ _('Project Home') }}</a>
+            <a href="//${ config.domains.dashboard }/projects/${ config.projects.current.slug }/">Project Home</a>
           </dd>
           <dd>
-            <a href="//{{ PRODUCTION_DOMAIN }}/builds/{{ slug }}/?fromdocs={{ slug }}">{{ _('Builds') }}</a>
+            <a href="//${ config.domains.dashboard }/projects/${ config.projects.current.slug }/builds/">Builds</a>
           </dd>
       </dl>
     </div>
   </div>
-{% endif %}
+          `;
+
+     document.body.insertAdjacentHTML("beforeend", flyout);
+
+     {# TODO: disable the flyout auto-injected by `readthedocs-addons.js` somehow #}
+     document.querySelector("readthedocs-flyout").remove();
```

Comment:
I thought that we had landed on giving theme authors the ability to include and element like:

`<readthedocs-flyout disabled></readthedocs-flyout>`

Or something similar. This probably makes the most sense: it doesn't require mucking with async code, and it doesn't give theme authors undue control over addons configuration.

Reply:
I need to remove the `readthedocs-flyout` here because it's injected by default via the addons. We need to find a way to tell the addons library to not do that when the users want to use the JSON data by themselves.

We haven't talk about the `disabled` attribute before, but probably is a good idea to communicate exactly what I'm needing here 👍🏼 

I will explore this and hopefully have something working that does not requires removing an HTML element as I'm doing now.

Reply:
> We need to find a way to tell the addons library to not do that when the users want to use the JSON data by themselves.

Yup, I understood that part. I thought we already talked about using our elements in this way though. Theme maintainers would use a definition of our elements to have some control of the flyout, which would include the ability to disable automatic injection the flyout.

> We haven't talk about the disabled attribute before

This attribute is just an example too, I don't have strong opinions on `disabled` specifically. I don't think we talked about the specific attribute, just that this would be the pattern for theme maintainers to use to disable some of our elements/features. I'm not finding the discussion though.

Reply:
I removed this line for now. We won't be automatically disabling the _official_ flyout injected for now. This is the safest decision for now and we can continue discussing this in another issue/pr in the near future.